src/Altinn.App.Api/Helpers/RequestHandling/DataRestrictionValidation.cs

```diff
@@ -19,77 +18,126 @@ public static class DataRestrictionValidation
         /// </summary>
         /// <param name="request">the original http request</param>
         /// <param name="dataType">datatype the files is beeing uploaded to</param>
-        /// <param name="errorResponse">Null if validation passed, error response if not</param>
         /// <returns>true with errorResponse = null if all is ok, false with errorResponse if not</returns>
-        public static bool CompliesWithDataRestrictions(HttpRequest request, DataType? dataType, out ActionResult? errorResponse)
+        public static (bool Success, List<ValidationIssue> Errors) CompliesWithDataRestrictions(HttpRequest request, DataType? dataType)
```

Comment:
Mostly to be compatible with how it was previously when validating data restrictions since we then returned immediately with one error. It was changed to list to support multiple errors as part of the new file analysis feature. Suggest we keep it like this in this round, an change the behavior when we change the result to be of type ProblemDetails/ValidationProblem.